src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/TensorPrimitives.Helpers.cs

```diff
@@ -23,7 +25,21 @@ private static void ValidateInputOutputSpanNonOverlapping<T>(ReadOnlySpan<T> inp
         }
 
         /// <summary>Throws an <see cref="OverflowException"/> for trying to negate the minimum value of a two-complement value.</summary>
-        internal static void ThrowNegateTwosCompOverflow() => throw new OverflowException(SR.Overflow_NegateTwosCompNum);
+        private static void ThrowNegateTwosCompOverflow() => throw new OverflowException(SR.Overflow_NegateTwosCompNum);
+
+        /// <summary>Creates a span of <typeparamref name="TTo"/> from a <typeparamref name="TFrom"/> when they're the same type.</summary>
+        private static unsafe Span<TTo> Rename<TFrom, TTo>(Span<TFrom> span) // MemoryMarshal.Cast, but without constraints or validation
+        {
+            Debug.Assert(sizeof(TFrom) == sizeof(TTo));
+            return MemoryMarshal.CreateSpan(ref Unsafe.As<TFrom, TTo>(ref MemoryMarshal.GetReference(span)), span.Length);
```

Comment:
Would it make sense to use `MemoryMarshal.Cast` and depend on the JIT to optimize it?

Reply:
I can't. MemoryMarshal.Cast constrains its type arguments to be structs.

Reply:
Ok, just read the comment above..

Reply:
I wonder whether there is a case for `internal static Span<TTo> CastUnsafe<TFrom, TTo>(Span<TFrom> span)`.

Reply:
in MemoryMarshal

Reply:
We can also just drop the constraints and replace them with dynamic check - like what's proposed in https://github.com/dotnet/runtime/issues/99205